### PR TITLE
Feature/implement ut for get interior vehicle data request

### PR DESCRIPTION
**Pull request by AByzhynar**
- Created base structure and mocks for RC command unit-tests
- Added constness to `Message` operator `==`
- Added UT for GetInteriorVehicleData class methods
- Added ogging to BaseCommandRequest class methods

### CODE REVIEW

File: src/components/can_cooperation/test/commands/get_interior_vehicle_data_request_test.cc [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+    ServicePtr exp_service(mock_service_);
+    mock_module_.set_service(exp_service);
+  }
+
+  can_cooperation::request_controller::MobileRequestPtr CreateCommand(
+      application_manager::MessagePtr msg) {
+    return can_cooperation::RCCommandFactory::CreateCommand(msg, mock_module_);
+  }
+
+  void HandleMessage() {
```
Comment (LuxoftAKutsan) [inline src/components/can_cooperation/test/commands/get_interior_vehicle_data_request_test.cc]: @AByzhynar remove this function

File: src/components/can_cooperation/test/commands/get_interior_vehicle_data_request_test.cc [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+    ON_CALL(mock_module_, service()).WillByDefault(Return(mock_service_));
+    ON_CALL(*mock_service_, GetApplication(app_id_))
+        .WillByDefault(Return(mock_app_));
+    EXPECT_CALL(mock_module_, event_dispatcher())
+        .WillRepeatedly(ReturnRef(event_dispatcher_));
+    ServicePtr exp_service(mock_service_);
+    mock_module_.set_service(exp_service);
+  }
+
+  can_cooperation::request_controller::MobileRequestPtr CreateCommand(
```
Comment (LuxoftAKutsan) [inline src/components/can_cooperation/test/commands/get_interior_vehicle_data_request_test.cc:122]: @AByzhynar Create funciton `CreateMessage` and make message as stack variable in test case

File: src/components/can_cooperation/test/commands/get_interior_vehicle_data_request_test.cc [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+};
+
+TEST_F(GetInteriorVehicleDataRequestTest,
+       Execute_MessageValidationOk_ExpectCorrectMessageSentToHMI) {
+  message_->set_connection_key(kConnectionKey);
+  message_->set_function_id(MobileFunctionID::GET_INTERIOR_VEHICLE_DATA);
+  message_->set_function_name("GetInteriorVehicleData");
+  message_->set_json_message(kValidMobileRequest);
+  EXPECT_CALL(mock_module_, service()).WillOnce(Return(mock_service_));
+  EXPECT_CALL(*mock_service_, GetApplication(_)).WillOnce(Return(mock_app_));
```
Comment (LuxoftAKutsan) [inline src/components/can_cooperation/test/commands/get_interior_vehicle_data_request_test.cc]: @AByzhynar specify app_id for application

File: src/components/can_cooperation/test/commands/get_interior_vehicle_data_request_test.cc [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+  can_cooperation::request_controller::MobileRequestPtr command =
+      CreateCommand(message_);
+  EXPECT_CALL(*mock_service_, ValidateMessageBySchema(*message_))
+      .WillOnce(Return(application_manager::MessageValidationResult::SUCCESS));
+  EXPECT_CALL(*mock_service_, CheckPolicyPermissions(_))
```
Comment (LuxoftAKutsan) [inline src/components/can_cooperation/test/commands/get_interior_vehicle_data_request_test.cc]: @AByzhynar What argument retries CheckPolicyPermissions? ie is possible to verify it?

File: src/components/can_cooperation/test/commands/get_interior_vehicle_data_request_test.cc [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+      .WillOnce(Return(mobile_apis::Result::eType::SUCCESS));
+  application_manager::AppExtensionPtr invalid_ext;
+  EXPECT_CALL(*mock_app_, QueryInterface(kModuleId))
+      .WillOnce(Return(invalid_ext))
+      .WillRepeatedly(Return(can_app_extention_));
+  EXPECT_CALL(*mock_app_, AddExtension(_)).WillOnce(Return(true));
```
Comment (LuxoftAKutsan) [inline src/components/can_cooperation/test/commands/get_interior_vehicle_data_request_test.cc]: @AByzhynar check parameter

File: src/components/can_cooperation/test/commands/get_interior_vehicle_data_request_test.cc [hunk 6]
```diff
@@ -0,0 +1,266 @@
...
+  message_->set_json_message(kValidMobileRequest);
+  EXPECT_CALL(mock_module_, service()).WillOnce(Return(mock_service_));
+  EXPECT_CALL(*mock_service_, GetApplication(_)).WillOnce(Return(mock_app_));
+  can_cooperation::request_controller::MobileRequestPtr command =
+      CreateCommand(message_);
+  EXPECT_CALL(*mock_service_, ValidateMessageBySchema(*message_))
+      .WillOnce(Return(application_manager::MessageValidationResult::SUCCESS));
+  EXPECT_CALL(*mock_service_, CheckPolicyPermissions(_))
+      .WillOnce(Return(mobile_apis::Result::eType::SUCCESS));
+  application_manager::AppExtensionPtr invalid_ext;
```
Comment (LuxoftAKutsan) [inline src/components/can_cooperation/test/commands/get_interior_vehicle_data_request_test.cc]: @AByzhynar please move all expectation after act

**Review by LuxoftAKutsan — CHANGES_REQUESTED** [review]

**Review by LitvinenkoIra — APPROVED** [review]

File: src/components/can_cooperation/test/commands/get_interior_vehicle_data_request_test.cc (hunk 2 above)
Comment (AByzhynar) [reply src/components/can_cooperation/test/commands/get_interior_vehicle_data_request_test.cc:122]: @LuxoftAKutsan Done in 654bc1b38ae6460f980a99f33991a07d87deb38d

File: src/components/can_cooperation/test/commands/get_interior_vehicle_data_request_test.cc (hunk 1 above)
Comment (AByzhynar) [reply src/components/can_cooperation/test/commands/get_interior_vehicle_data_request_test.cc]: @LuxoftAKutsan Done in 654bc1b38ae6460f980a99f33991a07d87deb38d

File: src/components/can_cooperation/test/commands/get_interior_vehicle_data_request_test.cc (hunk 3 above)
Comment (AByzhynar) [reply src/components/can_cooperation/test/commands/get_interior_vehicle_data_request_test.cc]: @LuxoftAKutsan Done in 654bc1b38ae6460f980a99f33991a07d87deb38d

File: src/components/can_cooperation/test/commands/get_interior_vehicle_data_request_test.cc (hunk 4 above)
Comment (AByzhynar) [reply src/components/can_cooperation/test/commands/get_interior_vehicle_data_request_test.cc]: @LuxoftAKutsan Done in 654bc1b38ae6460f980a99f33991a07d87deb38d

File: src/components/can_cooperation/test/commands/get_interior_vehicle_data_request_test.cc (hunk 6 above)
Comment (AByzhynar) [reply src/components/can_cooperation/test/commands/get_interior_vehicle_data_request_test.cc]: @LuxoftAKutsan Done in 654bc1b38ae6460f980a99f33991a07d87deb38d

File: src/components/can_cooperation/test/commands/get_interior_vehicle_data_request_test.cc (hunk 5 above)
Comment (AByzhynar) [reply src/components/can_cooperation/test/commands/get_interior_vehicle_data_request_test.cc]: @LuxoftAKutsan Done in 654bc1b38ae6460f980a99f33991a07d87deb38d

**Review by LuxoftAKutsan — APPROVED** [review]

**Pull request merged**